core/arxiv/submission/domain/event/__init__.py

```diff
@@ -0,0 +1,1263 @@
+        if submission.version == 1:
+            submission.status = Submission.DELETED
+            return submission
+        submission.version -= 1
+        target = submission.versions[-1]
+        # Return these to last announced state.
+        submission.status = target.status
+        submission.source_content = target.source_content
+        submission.submitter_contact_verified = \
+            target.submitter_contact_verified
+        submission.submitter_accepts_policy = \
+            target.submitter_accepts_policy
+        submission.submitter_confirmed_preview = \
+            target.submitter_confirmed_preview
+        submission.license = target.license
+        submission.metadata = copy.deepcopy(target.metadata)
+        return submission
+
+
+@dataclass(init=False)
+class ConfirmContactInformation(Event):
+    """Submitter has verified their contact information."""
+
+    NAME = "confirm contact information"
+    NAMED = "contact information confirmed"
+
+    def validate(self, submission: Submission) -> None:
+        """Cannot apply to a finalized submission."""
+        validators.submission_is_not_finalized(self, submission)
+
+    def project(self, submission: Submission) -> Submission:
+        """Update :attr:`.Submission.submitter_contact_verified`."""
+        submission.submitter_contact_verified = True
+        return submission
+
+
+@dataclass()
+class ConfirmAuthorship(Event):
+    """The submitting user asserts whether they are an author of the paper."""
+
+    NAME = "confirm that submitter is an author"
+    NAMED = "submitter authorship status confirmed"
+
+    submitter_is_author: bool = True
+
+    def validate(self, submission: Submission) -> None:
+        """Cannot apply to a finalized submission."""
+        validators.submission_is_not_finalized(self, submission)
+
+    def project(self, submission: Submission) -> Submission:
+        """Update the authorship flag on the submission."""
+        submission.submitter_is_author = self.submitter_is_author
+        return submission
+
+
+@dataclass(init=False)
+class ConfirmPolicy(Event):
+    """The submitting user accepts the arXiv submission policy."""
+
+    NAME = "confirm policy acceptance"
+    NAMED = "policy acceptance confirmed"
+
+    def validate(self, submission: Submission) -> None:
+        """Cannot apply to a finalized submission."""
+        validators.submission_is_not_finalized(self, submission)
+
+    def project(self, submission: Submission) -> Submission:
+        """Set the policy flag on the submission."""
+        submission.submitter_accepts_policy = True
+        return submission
+
+
+@dataclass()
+class SetPrimaryClassification(Event):
+    """Update the primary classification of a submission."""
+
+    NAME = "set primary classification"
+    NAMED = "primary classification set"
+
+    category: Optional[taxonomy.Category] = None
+
+    def validate(self, submission: Submission) -> None:
```

Comment:
We may need to do some additional validation on general categories like physics.gen-ph. I'll need to verify this.